eras/conway/impl/test/Test/Cardano/Ledger/Conway/RatifySpec.hs

```diff
@@ -0,0 +1,95 @@
+
+spec :: Spec
+spec = do
+  describe "Ratification" $ do
+    drepsProp @Conway
+
+drepsProp :: forall era. Era era => Spec
+drepsProp =
+  prop "DRep vote counts" $
+    forAll (arbitrary @(Map (DRep (EraCrypto era)) (CompactForm Coin))) $
+      \dRepDistr -> do
+        forAll (shuffle (Map.keys dRepDistr)) $ \dreps -> do
+          let size = fromIntegral $ length dreps
+              yes = ratio 30 size
+              drepsYes = onlyDrepsCred $ take yes dreps
+              votesYes = Map.fromList $ [(cred, VoteYes) | DRepCredential cred <- drepsYes]
+              CompactCoin stakeYes = fold $ Map.restrictKeys dRepDistr (Set.fromList drepsYes)
+
+              no = ratio 40 size
+              drepsNo = onlyDrepsCred $ take no . drop yes $ dreps
+              votesNo = Map.fromList $ [(cred, VoteNo) | DRepCredential cred <- drepsNo]
+              CompactCoin stakeNo = fold $ Map.restrictKeys dRepDistr (Set.fromList drepsNo)
+
+              abstain = ratio 10 size
+              drepsAbstain = onlyDrepsCred $ take abstain . drop (yes + no) $ dreps
+              votesAbstain = Map.fromList [(cred, Abstain) | DRepCredential cred <- drepsAbstain]
+              CompactCoin stakeAbstain = fold $ Map.restrictKeys dRepDistr (Set.fromList drepsAbstain)
+
+              CompactCoin stakeAlwaysAbstain = fromMaybe (CompactCoin 0) $ Map.lookup DRepAlwaysAbstain dRepDistr
+              CompactCoin stakeAlwaysNoConfidence = fromMaybe (CompactCoin 0) $ Map.lookup DRepAlwaysNoConfidence dRepDistr
+              CompactCoin notVotedStake =
+                fold $
+                  Map.withoutKeys
+                    dRepDistr
+                    (Set.fromList (drepsYes ++ drepsNo ++ drepsAbstain ++ [DRepAlwaysAbstain, DRepAlwaysNoConfidence]))
+
+              votes = Map.union votesYes $ Map.union votesNo votesAbstain
+
+              CompactCoin totalStake = fold dRepDistr
+
+              actual = dRepAcceptedRatio @era dRepDistr votes InfoAction
+              expected
+                | totalStake == stakeAbstain + stakeAlwaysAbstain = Nothing
+                | otherwise = Just $ stakeYes % (totalStake - stakeAbstain - stakeAlwaysAbstain)
+
+          actual `shouldBe` expected
+
+          let expectedRephrased
+                | stakeYes + stakeNo + notVotedStake + stakeAlwaysNoConfidence == 0 = Nothing
+                | otherwise = Just $ stakeYes % (stakeYes + stakeNo + notVotedStake + stakeAlwaysNoConfidence)
+          actual `shouldBe` expectedRephrased
+
+          let actualNoConfidence = dRepAcceptedRatio @era dRepDistr votes (NoConfidence SNothing)
+              expectedNoConfidence
+                | totalStake == stakeAbstain + stakeAlwaysAbstain = Nothing
+                | otherwise = Just $ (stakeYes + stakeAlwaysNoConfidence) % (totalStake - stakeAbstain - stakeAlwaysAbstain)
+          actualNoConfidence `shouldBe` expectedNoConfidence
+  where
+    ratio pct tot = ceiling $ fromRational ((pct * tot) % 100)
```

Comment:
This will allow you to remove the ghc warning flag at the top
```suggestion
    ratio pct tot = ceiling $ (pct * tot) % 100
```

Reply:
Thanks!